apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/DbScanVerdictStore.scala

```diff
@@ -0,0 +1,435 @@
+                    case None =>
+                      DBIO.failed(new RuntimeException("insertVerdict did not return row_id"))
+                  }
+                  views = mkViews(rowId)
+                  _ <- DBIO.sequence(views.map(sqlInsertView)).map(_ => ())
+                } yield ()
+              })
+              .map(_ => ())
+          } else DBIO.successful(())
+      } yield ()
+
+      futureUnlessShutdownToFuture(
+        storage.queryAndUpdate(
+          action.transactionally,
+          "scanVerdict.insertVerdictAndTransactionViews.batch",
+        )
+      )
+    }
+  }
+
+  def getVerdictByUpdateId(updateId: String)(implicit
+      tc: TraceContext
+  ): Future[Option[VerdictT]] = {
+    storage
+      .querySingle(
+        sql"""
+            select
+              row_id,
+              migration_id,
+              domain_id,
+              record_time,
+              finalization_time,
+              submitting_participant_uid,
+              verdict_result,
+              mediator_group,
+              update_id,
+              submitting_parties,
+              transaction_root_views
+            from #${Tables.verdicts}
+            where update_id = $updateId
+            limit 1
+          """.as[VerdictT].headOption,
+        "scanVerdict.getVerdictByUpdateId",
+      )
+      .value
+  }
+
+  private def afterFilters(
+      afterO: Option[(Long, CantonTimestamp)],
+      includeImportUpdates: Boolean,
+  ): NonEmptyList[SQLActionBuilder] = {
+    val gt = if (includeImportUpdates) ">=" else ">"
+    afterO match {
+      case None =>
+        NonEmptyList.of(sql"migration_id >= 0 and record_time #$gt ${CantonTimestamp.MinValue}")
+      case Some((afterMigrationId, afterRecordTime)) =>
+        NonEmptyList.of(
+          sql"migration_id = ${afterMigrationId} and record_time > ${afterRecordTime} ",
+          sql"migration_id > ${afterMigrationId} and record_time #$gt ${CantonTimestamp.MinValue}",
+        )
+    }
+  }
+
+  private def verdictsQuery(
+      filters: NonEmptyList[SQLActionBuilder],
+      orderBy: SQLActionBuilder,
+      limit: Int,
+  ) = {
+    def makeSubQuery(afterFilter: SQLActionBuilder) = {
+      sql"""
+      (select
+        row_id,
+        migration_id,
+        domain_id,
+        record_time,
+        finalization_time,
+        submitting_participant_uid,
+        verdict_result,
+        mediator_group,
+        update_id,
+        submitting_parties,
+        transaction_root_views
+      from #${Tables.verdicts}
+      where """ ++ afterFilter ++
+        sql" order by " ++ orderBy ++ sql" limit $limit)"
+    }
+
+    if (filters.size == 1) makeSubQuery(filters.head)
+    else {
+      val unionAll = filters.map(makeSubQuery).reduceLeft(_ ++ sql" union all " ++ _)
+      sql"select * from (" ++ unionAll ++ sql") all_queries " ++
+        sql"order by " ++ orderBy ++ sql" limit $limit"
+    }
+  }
+
+  def listVerdicts(
```

Comment:
This is here just use with the `getEventsReference` implementation, and will not be used by the proper SQL based query, which will read multiple tables(to be implemented).